app/src/main/java/com/github/dfa/diaspora_android/web/CustomWebViewClient.java

```diff
@@ -56,6 +57,11 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
                 || (host != null && (url.startsWith("https://" + host)
                 || url.startsWith("http://" + host)))) {
             return false;
+        }//make youtube links open external-->never customtab
+        if (url.startsWith("https://youtube")||url.startsWith("https://www.youtube")) {
+            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+            view.getContext().startActivity(intent);
```

Comment:
Can you do any checks here if there **really** is some youtube app installed? In case there is none the video should be still opened in customtab.

For this you can check against application-ids of official youtube app, newpipe, those you can find at F-Droid and any else comes to your mind.
For checking if there is a matching intent, you can use code found [here](https://github.com/gsantner/markor/blob/master/app/src/main/java/net/gsantner/opoc/util/ShareUtil.java#L786). Change application IDs in array and if `if pm.resolveService` one time gets true, then you know if theres an matching app installed.

Reply:
you can check if app is installed. I have some working code on Disroot app:

`String newpipe = "org.schabi.newpipe";
Intent tube = getPackageManager().getLaunchIntentForPackage(newpipe);`
`if(tube == !null) {
                    view.getContext().startActivity(intent);;
                }`

but i am not able to make it work on Dandelion*

Reply:
Checkout the `here` link :-)

Reply:
Ok, thanks for trying still. Maybe we should go the other way round:

As this can escape from the app (the user might be using proxy or Tor) I think an settings preference would be worth it. So if user set checked and url == youtube, then make intent.

What do you think about that?

Reply:
Great to be on the same road ;).

Of course, work on it when you can take time! Making option is quite easy, copy paste checkbox pref block, change texts and add AppSettings getter

Reply:
Thanks much for contributing to the peoject again, very welcome. I think it's an good idea to keep both projects active, and both do profit from changes of each :-).